Created an else-if.lcdoc file - Bug #19786

MartinKoob opened the pull request:
While working on bug #17794 changes to the If.lcdoc file I noticed there was no entry in the dictionary for the keyword "else if".  See - see Bug #19786

MartinKoob committed:
Created an else-if.lcdoc file - see Bug #19786

While working on bug #17794 changes to the If.lcdoc file I noticed there was no entry in the dictionary for the keyword "else if"

MartinKoob commented:
hmmm.... with the docs-helper.livecode stack I can test this file for errors and it shows 'no errors'  but the Preview does not work.  Is this because it is a new file?  

Also when I marked up 'else if' as a keyword in the if.lcdoc file in PR #5546 it did not format them as references in the preview of the document.

Is there also a document that lists all of the dictionary entries that needs to be updated when adding a new entry so it will be properly cross-referenced or is this built on the fly?

livecodeali commented on docs/dictionary/keyword/else-if.lcdoc:
```diff
@@ -0,0 +1,35 @@
+
+Type: keyword
+
+Syntax: else if
```
I think the syntax should maybe be
```
else if <condition> then
   <statementList>
```

livecodeali replied:
and then copy the `condition:` and `statenentList:` parameter descriptions from `if.lcdoc` once that's done.

livecodeali commented on docs/dictionary/keyword/else-if.lcdoc:
```diff
@@ -0,0 +1,35 @@
+Syntax: else if
+
+Summary:
+Used within an <if> <control structure> to evaluate an alternate <conditional expression|condition> and contain
```
this should be `<condition|conditional expression>` - so that it refers to the condition parameter but is rendered as 'conditional expression'.

livecodeali commented on docs/dictionary/keyword/else-if.lcdoc:
```diff
@@ -0,0 +1,35 @@
+
+Summary:
+Used within an <if> <control structure> to evaluate an alternate <conditional expression|condition> and contain
+a <statement|statement> or <statementlist|Statement List> to be executed if it's
```
`<A|A>` is equivalent to `<A>` in the docs markup, i.e. this can just be `<statement>` (as long as it has a reference below - see comment on references)

livecodeali replied:
Also `statement list` doesn't need capitalisation here

livecodeali replied:
Should be `its` not `it's`

livecodeali commented on docs/dictionary/keyword/else-if.lcdoc:
```diff
@@ -0,0 +1,35 @@
+Summary:
+Used within an <if> <control structure> to evaluate an alternate <conditional expression|condition> and contain
+a <statement|statement> or <statementlist|Statement List> to be executed if it's
+<conditional expression|condition> is true.
```
again this should be `<condition|conditional expression>`

livecodeali commented on docs/dictionary/keyword/else-if.lcdoc:
```diff
@@ -0,0 +1,35 @@
+end if
+
+Description:
+Use the <else if><keyword> to add an alternate <conditional expression|condition> to be evaluated in the
```
`<condition|conditional expression>`

livecodeali commented on docs/dictionary/keyword/else-if.lcdoc:
```diff
@@ -0,0 +1,35 @@
+
+Description:
+Use the <else if><keyword> to add an alternate <conditional expression|condition> to be evaluated in the
+<if><control structure>. If the <conditional expression|condition> evaluates to true then the <statment> or
```
missing an 'e' in `<statement>`

MartinKoob committed:
Updated else-if.lcdoc change syntax, add params

Implemented recommendations adding the changes to the syntax and the parameters.

MartinKoob committed:
Update else-if.lcdoc re recommended changes

Incorporating recommended changes into the document.

MartinKoob committed:
Create bug fix-17794.md

created new file to add the 'else if' keyword

livecodeali commented on docs/dictionary/keyword/else-if.lcdoc:
```diff
@@ -0,0 +1,50 @@
+
+References: if (control structure), break (control structure),
+keyword (glossary), control structure (glossary), then (keyword),
+end (keyword), else (keyword), control (keyword)
```
Oops, sorry missed this before - the references list is missing `switch (control structure), repeat (control structure), try (control structure)`

MartinKoob committed:
Update else-if.lcdoc add missing references

added references switch (control structure), repeat (control structure), try (control structure)

MartinKoob commented:
Not sure if I am supposed to respond to reviewer comments but did anyway.  Let me know if any other changes needed.

livecodeali commented:
@MartinKoob Looks good, thanks!
@livecode-vulcan review ok 166206f

MartinKoob committed:
Rename bugfix-17794.md to bugfix-19786.md

livecodeali commented:
Hah, well spotted!
@livecode-vulcan review ok 3250462

livecode-vulcan commented:
💙 review by @livecodeali ok 32504627cd072211c6f3b6ae4ba9fde197e9ea87

livecode-vulcan commented:
😎 test success 32504627cd072211c6f3b6ae4ba9fde197e9ea87

* try-community-armv6-android-api8: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-api8/builds/720)
* try-community-armv6-android-api9: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-api9/builds/736)
* try-community-js-emscripten-sdk1.35: [success](https://vulcan.livecode.com/builders/try-community-js-emscripten-sdk1.35/builds/710)
* try-community-universal-ios-iphoneos10.3: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphoneos10.3/builds/653)
* try-community-universal-ios-iphonesimulator10.3: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphonesimulator10.3/builds/656)
* try-community-universal-mac-macosx10.6: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.6/builds/714)
* try-community-universal-mac-macosx10.9: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.9/builds/734)
* try-community-x86-linux-debian7: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian7/builds/719)
* try-community-x86-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian8/builds/736)
* try-community-x86_64-linux-debian7: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian7/builds/718)
* try-community-x86_64-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian8/builds/729)
* try-community-x86-win32: [success](https://vulcan.livecode.com/builders/try-community-x86-win32/builds/751)
* try-community-x86_64-win32: [success](https://vulcan.livecode.com/builders/try-community-x86_64-win32/builds/715)

MartinKoob commented:
The else-if bug report for this PR http://quality.livecode.com/show_bug.cgi?id=19786 has its status set as 'hibernated'.  Should it be changed to 'awaiting release'?

livecodepanos commented:
Good spot @MartinKoob, my bad, sorry :)

I'll change it now.
